runtime/gc_base/modronapi.cpp

```diff
@@ -586,7 +586,11 @@ j9gc_pool_maxmemory(J9JavaVM *javaVM, UDATA poolID)
 		maxsize = extensions->memoryMax;
 		break;
 	case J9_GC_MANAGEMENT_POOL_REGION_EDEN :
-		maxsize = extensions->tarokIdealEdenMaximumBytes;
+		if (extensions->userSpecifiedParameters._Xmn._wasSpecified || extensions->userSpecifiedParameters._Xmnx._wasSpecified) {
+			maxsize = extensions->tarokIdealEdenMaximumBytes;
```

Comment:
We should not be using things associated with cmdline opiton like _Xmn._wasSpecified here, but rely on parameters (other than tarokIdealEdenMaximumBytes) that already account for those cmdline options.

Specifically, we could use _maxEdenRegionCount, instead.

Ideally, we should try to simplify the code so that  tarokIdealEdenMaximumBytes and  _maxEdenRegionCount are in sync, or we can even try remove one of them (since it's easy to derive one from another, when/if needed)

Some guidelines for Balanced that I believe hold now and we should preserve:
- default initial Eden/Total ratio is 1/4 (if nothing or Xms or Xmx are specified, but no Xmn or Xmo options are specified)
- default maximum and minimum  Eden/Total ratio is 3/4 and 0/4
- if someone specifies Xmn(x,s) options, they should be used to calculate new max,min Eden/Total ratios
- (there could be some complicated variants when both Xmn(x,s) and Xmo(x,s) options are specified, that I did not think of yet, but for the most part Xmo options should not affect Eden initial/max/min sizing).

Reply:
I am trying to sync _maxEdenRegionCount and extensions->tarokIdealEdenMaximumBytes after MM_SchedulingDelegate._maxEdenRegionCount is initialized, tarokIdealEdenMaximumBytes has been used in a couple of other places, checking if the sync would impact some behaviours. 
1, tarokIdealEdenMaximumBytes was used to set tarokAllocationAgeUnit in MM_IncrementalGenerationalGC::initialize()
2, tarokMaximumNurseryAgeInBytes was initialized with tarokIdealEdenMaximumBytes*2 in MM_IncrementalGenerationalGC::initialize().